rust/template/d3log/src/value_set.rs

```diff
@@ -0,0 +1,117 @@
+// module to wrap the general encoding type DDValue for rust compiled ddlog
+
+use crate::{Batch, BatchBody, Error, Evaluator};
+use differential_datalog::{ddval::DDValue, program::RelId, DeltaMap};
+use std::collections::{BTreeMap, HashMap};
+use std::fmt;
+use std::fmt::Display;
+use std::sync::{Arc, Mutex};
+
+#[derive(Clone)]
+pub struct ValueSet {
+    pub eval: Evaluator, // used to translate to record
+    pub deltas: Arc<Mutex<DeltaMap<differential_datalog::ddval::DDValue>>>,
+}
+
+impl Display for ValueSet {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(&"<")?;
+        let iter = self.clone();
+        let mut m: usize = 0;
+        for (relid, _v, _w) in iter.into_iter() {
+            // I would really prefer a readable name..but we need an Evaluator and it doesn't seem right
+            // to add it to all the batches or globalize it
+            f.write_str(&format!("({}", relid))?;
+            m += 1;
+            f.write_str(&format!(" {})", m))?;
+        }
+        f.write_str(&">")?;
+        Ok(())
```

Comment:
The reference is unnecessary here btw, are you using [clippy](https://github.com/rust-lang/rust-clippy)?
```suggestion
        f.write_char('>')
```